execution/stagedsync/stage_mining_exec.go

```diff
@@ -127,8 +127,12 @@ func SpawnMiningExecStage(s *StageState, txc wrap.TxContainer, cfg MiningExecCfg
 	}
 	defer sd.Close()
 
+	chainReader := ChainReaderImpl{config: cfg.chainConfig, tx: txc.Tx, blockReader: cfg.blockReader, logger: logger}
+
 	txNum := sd.TxNum()
 
+	core.InitializeBlockExecution(cfg.engine, chainReader, current.Header, cfg.chainConfig, ibs, &state.NoopWriter{}, logger, nil)
```

Comment:
You can pass `nil` instead of `NoopWriter` here,  there is logic inside the function to use `NoopWriter` if stater writer is nil.